File: android_samples_java/02 App_basic_CallBackExamples/app/src/main/java/com/msnegi/callbackexamples/MessageReceiver.java
package com.msnegi.callbackexamples;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

public class MessageReceiver extends BroadcastReceiver {

    private CallBackInterface callback;

    public void setContext(CallBackInterface callback) {
        this.callback = callback;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        callback.callbackfunction("This is a callback from Broadcast receiver !!!");
    }
}

File: android_samples_kotlin/12 CallbackExamples/app/src/main/java/com/msnegi/callbackexamples/MessageReceiver.kt
package com.msnegi.callbackexamples

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class MessageReceiver : BroadcastReceiver() {
    private var callback: CallBackInterface? = null

    fun setContext(callback: CallBackInterface) {
        this.callback = callback
    }

    override fun onReceive(context: Context?, intent: Intent?) {
        callback!!.callbackfunction("This is a callback from Broadcast receiver !!!")
    }
}
